sdk/src/main/java/com/qonversion/android/sdk/internal/di/controllers/ControllersAssemblyImpl.kt

```diff
@@ -14,22 +21,33 @@ internal class ControllersAssemblyImpl(
     private val miscAssembly: MiscAssembly,
     private val servicesAssembly: ServicesAssembly
 ) : ControllersAssembly {
-    override val userPropertiesController: UserPropertiesController
-        get() = UserPropertiesControllerImpl(
-            pendingPropertiesStorage = storageAssembly.pendingUserPropertiesStorage,
-            sentPropertiesStorage = storageAssembly.sentUserPropertiesStorage,
-            service = servicesAssembly.userPropertiesService,
-            worker = miscAssembly.delayedWorker,
-            logger = miscAssembly.logger
+    override fun userPropertiesController(): UserPropertiesController =
+        UserPropertiesControllerImpl(
+            pendingPropertiesStorage = storageAssembly.pendingUserPropertiesStorage(),
+            sentPropertiesStorage = storageAssembly.sentUserPropertiesStorage(),
+            service = servicesAssembly.userPropertiesService(),
+            worker = miscAssembly.delayedWorker(),
+            logger = miscAssembly.logger()
         )
 
-    override fun getGoogleBillingController(purchasesListener: PurchasesListener): GoogleBillingController {
-        return GoogleBillingControllerImpl(
-            miscAssembly.googleBillingConsumer,
-            miscAssembly.googleBillingPurchaser,
-            miscAssembly.googleBillingDataFetcher,
+    override fun googleBillingController(purchasesListener: PurchasesListener): GoogleBillingController =
+        GoogleBillingControllerImpl(
+            provideGoogleBillingConsumer(),
+            provideGoogleBillingPurchaser(),
+            provideGoogleBillingDataFetcher(),
             purchasesListener,
-            miscAssembly.logger
+            miscAssembly.logger()
         )
-    }
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    fun provideGoogleBillingConsumer(): GoogleBillingConsumer =
+        GoogleBillingConsumerImpl(miscAssembly.logger())
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    fun provideGoogleBillingPurchaser(): GoogleBillingPurchaser =
+        GoogleBillingPurchaserImpl(miscAssembly.logger())
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    fun provideGoogleBillingDataFetcher(): GoogleBillingDataFetcher =
+        GoogleBillingDataFetcherImpl(miscAssembly.logger())
```

Comment:
names should be without the `provide` verb, as we discussed. I think we should stick to this rule even with non-interface build methods